src/mono/mono/mini/interp/transform.c

```diff
@@ -2515,11 +2551,52 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 		(!strncmp ("System.Runtime.Intrinsics.Arm", klass_name_space, 29) ||
 		!strncmp ("System.Runtime.Intrinsics.X86", klass_name_space, 29))) {
 		interp_generate_void_throw (td, MONO_JIT_ICALL_mono_throw_platform_not_supported);
-	} else if (in_corlib &&
-			   (!strncmp ("System.Numerics", klass_name_space, 15) &&
-				!strcmp ("Vector", klass_name) &&
-				!strcmp (tm, "get_IsHardwareAccelerated"))) {
-		*op = MINT_LDC_I4_0;
+	} else if (in_corlib && !strncmp ("System.Numerics", klass_name_space, 15)) {
+		if (!strcmp ("Vector", klass_name) &&
+				!strcmp (tm, "get_IsHardwareAccelerated")) {
+			*op = MINT_LDC_I4_0;
+		} else if (!strcmp ("BitOperations", klass_name)) {
+			if ((!strcmp (tm, "RotateLeft") || !strcmp (tm, "RotateRight")) && MINT_IS_LDC_I4 (td->last_ins->opcode)) {
+				gboolean left = !strcmp (tm, "RotateLeft");
+				int ct = interp_get_const_from_ldc_i4 (td->last_ins);
+				int opcode = -1;
+				gboolean is_i4 = (csignature->params [0]->type == MONO_TYPE_U4) || (csignature->params [0]->type == MONO_TYPE_U && SIZEOF_VOID_P == 4);
+				if (is_i4)
+					opcode = left ? MINT_ROL_I4_IMM : MINT_ROR_I4_IMM;
+				else
+					opcode = left ? MINT_ROL_I8_IMM : MINT_ROR_I8_IMM;
```

Comment:
should there be an assert here. This is making an assumption that there are only two versions `uint32` and `uint64` based (with `nuint` resolving to one of the two).

This would break and potentially not be caught if we added `byte`, `ushort`, `UInt128` or other variants.